streams/src/main/java/org/apache/kafka/streams/processor/internals/StreamsRebalanceListener.java

```diff
@@ -71,12 +72,7 @@ public void onPartitionsRevoked(final Collection<TopicPartition> partitions) {
             try {
                 taskManager.handleRevocation(partitions);
             } catch (final Throwable t) {
```

Comment:
Side remark: It's really not great to catch `Throwable` and reinterpret it, since it encompasses both Error and Exception. I.e., We might catch an `OutOfMemoryError` and wrap it as a `TaskMigratedException`, which would obviously be bad.

What do you think about changing these on the side to just catch `RuntimeException` or `Exception`?

Reply:
That sounds good

Reply:
Actually I think we should not interpret everything as `TaskMigrated` --> if it is due to a recoverable fenced issue e.g. the thrown exception would already be a TaskMigratedException. So I think we should not capture any here instead.

Reply:
I think `handleRevocation` cannot guarantee to catch all exceptions as TaskMigrated correct?

Reply:
I think Guozhang's point is that `handleRevocation` _might_ throw other exceptions (eg `IllegalStateException`) which we should not convert to `TaskMigratedException` because they are fatal while `TaskMigrated` is recoverable